main.tf

```diff
@@ -33,6 +41,14 @@ resource "aws_ec2_transit_gateway" "this" {
   )
 }
 
+resource "aws_ec2_tag" "this" {
+  count = var.create_tgw ? length(local.tgw_default_route_table_tags_merged) : 0
```

Comment:
`for_each` is generally a better option, because with `count` it will recreate tags when the order changes.

We should aim to use `for_each` for new features (like this one).

Reply:
agreed, however that would require making this resource *required* as opposed to optional. Since `count` and `for_each` are mutually exclusive.  Making this non-optional potentially opens to door to failed runs, since `aws_ec2_transit_gateway.this[0].association_default_route_table_id` is a dependency on of the resource, if `aws_ec2_transit_gateway.this` isn't created because someone passed the  `create_tgw` the module run would also fail. 

IMO If we want to do that we should refactor and remove the `create_tgw` parameter entirely.  But doing that seemed like it was outside the scope of this change.

Reply:
We can write `for_each` to be conditional, too. For example, https://github.com/terraform-aws-modules/terraform-aws-eventbridge/blob/master/main.tf#L166

`create_tgw` is important to control creation when calling this module from Terragrunt, for example.

Reply:
Hmm yeah that could work, let me add it to this branch and re-push!

Reply:
nice, that worked well! Just pushed an update!